src/pool.rs

```diff
@@ -34,6 +34,24 @@ impl<T: TaskCell + Send> ThreadPool<T> {
         self.remote.spawn(t);
     }
 
+    /// Scale workers of the thread pool, the new thread count should be less
+    /// than or equal to the max_thread_count, otherwise it will become no-op
+    /// and return false.
+    ///
+    /// If the pool is shutdown, it becomes no-op.
+    ///
+    /// Notice:
+    /// 1. The effect of scaling may be delayed, eg:
+    ///    - Threads run long-term tasks, resulting in inability to scale down in time,
+    ///      until they have no task to process and can sleep;
+    ///    - Scaling up relies on spawning tasks to the thread pool to unpark new threads,
+    ///      so the delay depends on the time interval between scale and spawn;
+    /// 2. If lots of tasks have been spawned before start, all threads (max_thread_count)
+    ///    will run until they can sleep, then scheduled based on core_thread_count;
+    pub fn scale_workers(&self, new_thread_count: usize) -> bool {
```

Comment:
Any tests?

Reply:
I only did system testing, I will think about adding unit testing.

Reply:
> I only did system testing, I will think about adding unit tests.

I had done three tests:
1. Spawn 10000 tasks first, then lazy build threads, from `pidstat -G yatp -t 1`, I can see in an instant `max_thread_count` threads are woken up to work and then drop to `core_thread_count` (I continue to spawn tasks);
2. Call scale_workers with `0`, `< min_thread_count`, `> max_thread_count`, I can see it will adjust to `max_thread_count`, `min_thread_count`, `max_thread_count`;
3. Continuous spawn task, call `scale_workers` every 3s, the parameter is a random number between 0~`max_thread_count`, observed `pidstat -G yatp -t 1` equal to scaled  thread number;

Do you think these are enough? Also, I don't think these test cases fit in tests.rs . eg: not easy to get the number of running threads.

Reply:
You can use blocking channels to test if all threads are woken up and block in unit tests.

Reply:
> You can use blocking channels to test if all threads are woken up and block in unit tests.

Added two test cases, thanks!